### PR TITLE
New payload

### PR DESCRIPTION
**Pull request by Ngoguey42**

### CODE REVIEW

**Comment by Ngoguey42** [comment]
This is now ready for review

File: src/irmin-pack/unix/file_manager.ml [hunk 1]
```diff
@@ -126,8 +127,27 @@ struct
     let* () = if t.use_fsync then Suffix.fsync t.suffix else Ok () in
     let* () =
       let pl : Payload.t = Control.payload t.control in
+      let status =
+        match pl.status with
+        | From_v1_v2_post_upgrade _ -> pl.status
+        | From_v3 ->
+            (* Using physical equality to test which indexing_strategy
+               we are using *)
+            if t.indexing_strategy == Irmin_pack.Indexing_strategy.minimal then
```
Comment (tomjridge) [inline src/irmin-pack/unix/file_manager.ml:136]: Phys equality a bit dangerous as we all know. I wonder if indexing_strategy should not be a function, but a tagged variant: Always | Minimal | Other of (...filter function...). Probably something for another time.

**Comment by tomjridge** [comment]
LGTM

**Review by icristescu — APPROVED** [review]
Thanks

**Pull request merged**